pkg/controller/operandrequest/reconcile_operand.go

```diff
@@ -153,70 +161,113 @@ func (r *ReconcileOperandRequest) createUpdateCr(service *operatorv1alpha1.Confi
 
 				unstruct.Object["spec"] = mergedCR
 				unstruct.Object["metadata"].(map[string]interface{})["namespace"] = namespace
-
+				if unstruct.Object["metadata"].(map[string]interface{})["labels"] == nil {
+					klog.V(4).Info("Adding ODLM label in the custom resource")
+					unstruct.Object["metadata"].(map[string]interface{})["labels"] = make(map[string]interface{})
+				}
+				unstruct.Object["metadata"].(map[string]interface{})["labels"].(map[string]interface{})["operator.ibm.com/opreq-control"] = t
 				// Creat or Update the CR
 				crCreateErr := r.client.Create(context.TODO(), &unstruct)
 				if crCreateErr != nil && !errors.IsAlreadyExists(crCreateErr) {
 					stateUpdateErr := r.updateServiceStatus(csc, service.Name, crdName, operatorv1alpha1.ServiceFailed)
 					if stateUpdateErr != nil {
+						klog.Error("Fail to update status")
 						merr.Add(stateUpdateErr)
 					}
 					klog.Error("Fail to Create the Custom Resource: ", crdName, ". Error message: ", crCreateErr)
 					merr.Add(crCreateErr)
-
 				} else if errors.IsAlreadyExists(crCreateErr) {
-					existingCR := &unstructured.Unstructured{
+					existingCR := unstructured.Unstructured{
 						Object: map[string]interface{}{
-							"apiVersion": unstruct.Object["apiVersion"].(string),
-							"kind":       unstruct.Object["kind"].(string),
+							"apiVersion": apiversion,
+							"kind":       kind,
 						},
 					}
-
 					crGetErr := r.client.Get(context.TODO(), types.NamespacedName{
-						Name:      unstruct.Object["metadata"].(map[string]interface{})["name"].(string),
+						Name:      name,
 						Namespace: namespace,
-					}, existingCR)
-
+					}, &existingCR)
 					if crGetErr != nil {
 						stateUpdateErr := r.updateServiceStatus(csc, service.Name, crdName, operatorv1alpha1.ServiceFailed)
 						if stateUpdateErr != nil {
+							klog.Error("Fail to update status")
 							merr.Add(stateUpdateErr)
 						}
 						klog.Error("Fail to Get the Custom Resource: ", crdName, ". Error message: ", crGetErr)
 						merr.Add(crGetErr)
 						continue
 					}
-					existingCR.Object["spec"] = unstruct.Object["spec"]
-					if crUpdateErr := r.client.Update(context.TODO(), existingCR); crUpdateErr != nil {
-						stateUpdateErr := r.updateServiceStatus(csc, service.Name, crdName, operatorv1alpha1.ServiceFailed)
+					if existingCR.Object["metadata"].(map[string]interface{})["labels"] != nil && existingCR.Object["metadata"].(map[string]interface{})["labels"].(map[string]interface{})["operator.ibm.com/opreq-control"] == t {
+						existingCR.Object["spec"] = unstruct.Object["spec"]
+						if crUpdateErr := r.client.Update(context.TODO(), &existingCR); crUpdateErr != nil {
+							stateUpdateErr := r.updateServiceStatus(csc, service.Name, crdName, operatorv1alpha1.ServiceFailed)
+							if stateUpdateErr != nil {
```

Comment:
code looks good, but it is better to separate the nested `if/for` into smaller functions.

Linux kernel code style: https://github.com/torvalds/linux/blob/master/Documentation/process/coding-style.rst#7-centralized-exiting-of-functions

> The answer to that is that if you need more than 3 levels of indentation, you're screwed anyway, and should fix your program.

Let's don't use more than 5 nested levels of `for/if`, the current nested level is 6. The less the better.

We can merge the code for now, and later enhance the logic.

Reply:
@chenzhiwei Yes, I plan to reflect the code in the reconcile operand, Currently, it is unreadable.